core/document/src/document.rs

```diff
@@ -300,6 +300,14 @@ impl Document {
 				let children = self.layer_panel(path.as_slice())?;
 				Some(vec![DocumentResponse::DocumentChanged, DocumentResponse::ExpandFolder { path, children }])
 			}
+			Operation::ToggleVisibility { path } => {
+				let _ = self.layer_mut(&path).map(|layer| {
+					layer.visible = !layer.visible;
+					layer.cache_dirty = true;
+				});
+				let children = self.layer_panel(&[])?;
```

Comment:
you should either update the folder the current layer resides (for that you can use `children = self.layer_panel(path.as_slice()[..path.len()-1])?`)
or you need to add a separate response for the vis toggle

Reply:
Should `children = self.layer_panel(path.as_slice()[..path.len()-1])?` replace line 308?

Reply:
you still need the `let` but yeah, you can try and see if it works